flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommittableCollectorSerializerTest.java

```diff
@@ -134,55 +190,78 @@ void testAlignSubtaskCommittableManagerCheckpointWithCheckpointCommittableManage
                 .isEqualTo(committableManager.getCheckpointId());
     }
 
+    /**
+     * @param assertMessageHeading prefix used for assertion fail message.
+     * @param subtaskId subtaskId to get {@link SubtaskCommittableManager} from {@link
+     *     CheckpointCommittableManagerImpl}
+     * @param expectedNumberOfSubtasks expected number of subtasks for {@link CommittableSummary}
+     * @param committableCollector collector to get {@link CheckpointCommittableManager}s from.
+     * @param expectedCommittableSize expected number of {@link CheckpointCommittableManager}.
+     * @param expectedNumberOfPendingRequestsPerCommittable expected number of pending request per
+     *     {@link SubtaskCommittableManager}.
+     */
     private void assertCommittableCollector(
             String assertMessageHeading,
             int subtaskId,
-            int numberOfSubtasks,
-            CommittableCollector<Integer> committableCollector) {
+            int expectedNumberOfSubtasks,
+            CommittableCollector<Integer> committableCollector,
+            int expectedCommittableSize,
+            Object[] expectedNumberOfPendingRequestsPerCommittable) {
```

Comment:
Why do you need the `Object[]` here? I would assume that in the test case, we know all types and do not need to use them. Please also avoid using arrays if possible.

Reply:
yeah, 
the thing is that depends on the test, `testCommittableCollectorV2SerDe` or `testCommittablesForSameSubtaskIdV2SerDe`

we will have `assertPendingRequests` will expect single Integer or List of integers, one for each restored committable for the same subtaskId.

I wanted to reuse the assertions from `testCommittableCollectorV2SerDe` which basically checks for the same things but with different expected values.

Im using Objects arrays because `assertThat(...).containsExactly(...) var args (single value or array)`.